nornir/core/inventory.py

```diff
@@ -439,18 +439,24 @@ def children_of_group(self, group: Union[str, Group]) -> Set[Host]:
                 hosts.add(host)
         return hosts
 
-    def add_host(self, name: str, defaults: Optional[Defaults] = None,
-                **kwargs) -> None:
+    def add_host(self, name: str, **kwargs) -> None:
         """
         Add a host to the inventory after initialization
         """
-        host = {name: Host(name, defaults=self.defaults, **kwargs)}
+        host = {
```

Comment:
Re-wrote `host` variable using the `deserialize_host` function rather than the direct `Host` class initialization